apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx

```diff
@@ -740,7 +740,7 @@ export default function DeviceAction<R, H extends States, P>({
   action: Action<R, H, P>;
   request: R;
 }): JSX.Element {
-  const device = useSelector(getCurrentDevice);
+  const device = useSelector(getCurrentDevice, shallowEqual);
```

Comment:
Can you explain the full reasoning for using `shallowEqual` here?
What's the reason for the reference of the in-store device to change ?
IMO it would be better to fix the issue at the source (what part of the code is updating the store with a different device object). So look at all redux dispatches and try to identify what's wrong.
This change impacts all device actions across all transports, and that needs to be properly assessed, and I think that making a fix specific to Speculos here might be a bit overkill, and it could have unwanted side effects.